packages/menu/src/SubMenu.tsx

```diff
@@ -242,80 +223,6 @@ const SubMenu: ExtendableBox<
 
     const numberOfMenuItems = React.Children.toArray(children).length;
 
-    const updatedGlyph =
-      glyph &&
-      React.cloneElement(glyph, {
-        role: 'presentation',
-        className: cx(
-          mainIconStyle,
-          {
-            [activeIconStyle]: active,
-            [focusStyles.iconStyle]: showFocus,
-          },
-          glyph.props?.className,
-        ),
-      });
-
-    const sharedBoxProps = {
-      ...subMenuContainer.prop,
-      ...rest,
-      ref,
-      onClick: onRootClick,
-      role: 'menuitem',
-      'aria-haspopup': true,
-      className: cx(
-        menuItemContainerStyle,
-        subMenuStyle,
-        linkStyle,
-        {
-          [activeMenuItemContainerStyle]: active,
-          [disabledMenuItemContainerStyle]: disabled,
-          [subMenuOpenStyle]: open,
-          [focusedMenuItemContainerStyle]: showFocus,
-        },
-        className,
-      ),
-    };
```

Comment:
Notable differences (that I should likely address):
 
`SubMenu`'s implementation includes `onClick` and `aria-haspopup` (while `MenuItem` does not)

`MenuItem`'s implementation includes `tabIndex`,  `aria-disabled` and `disabled`

Reply:
Additionally, `MenuItem` spreads `anchorProps` on the box, while `SubMenu` does not